reference/src/layout/pointers.md

```diff
@@ -2,34 +2,33 @@
 
 ### Terminology
 
-Reference types are types of the form `&T`, `&mut T` or `&dyn T`.
+Reference types are types of the form `&T`, `&mut T`.
 
 Raw pointer types are types of the form `*const T` or `*mut T`.
-We write `*T` when the mutability attribute is unimportant.
 
 ### Representation
 
 The alignment of `&T`, `&mut T`, `*const T` and `*mut T` are the same,
 and are at least the word size.
 
-* If `T` is a trait, then the alignment of `&dyn T` is the word size.
 * If `T` is a sized type then the alignment of `&T` is the word size.
+* The alignment of `&dyn Trait` is the word size.
 * The alignment of `&[T]` is the word size.
 * The alignment of `&str` is the word size.
 * Alignment in other cases may be more than the word size (e.g., for other dynamically sized types).
 
 The sizes of `&T`, `&mut T`, `*const T` and `*mut T` are the same,
 and are at least one word.
 
-* If `T` is a trait, then the size of `&dyn T` is two words.
 * If `T` is a sized type then the size of `&T` is one word.
+* The size of `&dyn Trait` is two words.
 * The size of `&[T]` is two words.
 * The size of `&str` is two words.
 * Size in other cases may be more than one word (e.g., for other dynamically sized types).
 
 ### Notes
 
-The layouts of `&T`, `&mut T` and `*T` are the same.
+The layouts of `&T`, `&mut T`, `*const T` and `*mut T` are the same.
```

Comment:
Did we never define whether "layout" is just size+align+field offset or whether it includes "niches for enum optimization"?

Reply:
I don't think we ever did (do we have a definition of layout somewhere?), but it makes sense to define niches as part of layout. Otherwise when we specify layout recursively for structs, etc. to depend on the layout of their fields, niches cannot play a role, but they do play a role for `Option` and that's already guaranteed.

Reply:
Then the sentence here is wrong though -- these types do not all have the same niche.

Reply:
That's right. I don't think this PR is the right place to fix that (this was already broken), but I've opened #122 to track that, and maybe we can discuss that in the next meeting.

Reply:
Fair.